### PR TITLE
Enables creating a garden through ui issue #726

### LINKED ISSUE #726: Need ability to create garden through ui
Creating garden through ui is need for parent and child communication, since parent cannot get child garden information without having an active stomp listener.

### PR DESCRIPTION
**Pull request by west270**
Enables creating a garden through ui.
Found new issue #727 while testing.
Fixes #726

### CODE REVIEW

File: src/ui/src/js/controllers/admin_garden.js [hunk 1]
```diff
@@ -19,13 +19,15 @@ export default function adminGardenController(
...
-
+  $scope.gardenCreateSchema = GardenService.CreateSCHEMA;
+   $scope.gardenCreateForm = GardenService.CreateFORM;
   $scope.successCallback = function(response) {
     $scope.response = response;
     $scope.data = response.data;
 
   };
-
+  $scope.garden_name = null;
```
Comment (TheBurchLog) [inline src/ui/src/js/controllers/admin_garden.js:29]: This is too generic for this page. The who page is to manage Gardens and having a field called `garden_name` gives no uniqueness to what the value is actually doing. Give this a more meaningful name.

File: src/ui/src/partials/admin_garden_index.html [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
   </span>
 </h1>
-  
+<div class="pull-right">
+  <div class="form-popup" id="myForm" ng-hide=createGardenFormHide>
+    <label><b>Garden Name</b></label>
+    <input type="text" placeholder="Enter Garden Name" name="garden_name" required ng-model="garden_name">
+
+    <button type="button" class="btn" ng-click="createGarden()">Create</button>
+    <button type="button" class="btn" ng-click="createGardenFormHide = true">Close</button>
```
Comment (TheBurchLog) [inline src/ui/src/partials/admin_garden_index.html:16]: On close, we should clear the value being set by the form. That way the user gets a clean form if they reopen it.

File: src/ui/src/js/controllers/admin_garden.js [hunk 3]
```diff
@@ -42,6 +44,19 @@ export default function adminGardenController(
...
+  $scope.createGarden = function() {
+      if ($scope.garden_name != "" & $scope.garden_name != null){
+        GardenService.createGarden({"name":$scope.garden_name, "status":"NOT_CONFIGURED"});
+        $state.go('base.garden_view',
+              {
+                'name': $scope.garden_name,
+              }
+            );
+      }
+      $scope.createGardenFormHide = true;
```
Comment (TheBurchLog) [inline src/ui/src/js/controllers/admin_garden.js:56]: Does this actually do anything? If this condition is hit, then the field was not populated, so isn't that an error condition? You should do an else statement and throw an validation alert that they didn't provide an Garden Name for the create function. Also, don't hide the form during the failure case.

**Review by TheBurchLog — COMMENTED** [review]
Once you address me comments this should be good ot go

**Comment by west270** [comment]
make lint:
```
flake8 beer_garden test
black --check beer_garden test
All done! ✨ 🍰 ✨
130 files would be left unchanged.
```


make test:
```
============================================================================================================================= warnings summary =============================================================================================================================
test/db/mongo/pruner_test.py: 12 warnings
  /bg_venv/venv/lib64/python3.7/site-packages/mongoengine/queryset/visitor.py:156: DeprecationWarning: 'empty' property is deprecated in favour of using 'not bool(filter)'
    warnings.warn(msg, DeprecationWarning)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================================================================================== 327 passed, 262 skipped, 12 warnings in 10.47s ==============================================================================================================
```

**Review by TheBurchLog — APPROVED** [review]

**Pull request merged**